### PR TITLE
new recipe: mtafsir

### PR DESCRIPTION
**Pull request by mazbrili**
qt application for koran interpretation browser

### CODE REVIEW

File: app-text/mtafsir/mtafsir-0.2.recipe [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	qmake
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/Mtafsir/{bin,share}
+	mkdir -p $dataDir/ayat
+	cp -rf ./usr/bin/mtafsir $appsDir/Mtafsir/bin/Mtafsir
+    cp -rf ./usr/share/* $appsDir/Mtafsir/share
```
Comment (korli) [inline app-text/mtafsir/mtafsir-0.2.recipe]:
```suggestion
	cp -rf ./usr/share/* $appsDir/Mtafsir/share
```

File: app-text/mtafsir/mtafsir-0.2.recipe [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+fi
+
+PROVIDES="
+	mtafsir$secondaryArchSuffix = $portVersion
+	app:Mtafsir = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	ayat_tafasir_data
+	lib:libGL$secondaryArchSuffix
```
Comment (extrowerk) [inline app-text/mtafsir/mtafsir-0.2.recipe]: Is it really needed?

Comment (mazbrili) [reply app-text/mtafsir/mtafsir-0.2.recipe]: @extrowerk  it is depend on ayat_tafasir_data recipe.. yes.. it contain data for mtafsir to avoid duplication

**Comment by extrowerk** [comment]
I meant libGL, which is not required or used at all.
You could clean up your recipe and the hpkg, as it ocntains plenty unneeded files and folders.
For example: why providing the icons if they never be used?  Why the main binary located in the /bin subfolder? Why the plenty rm? You could at least group them, like:
```
rm $appsDir/Mtafsir/share/mtafsir/tafasir/{tafasirInfo,baghawy.db,katheer.db,blablabla}
```
You could even eliminate the unneeded "share/mtafsir/" subfolder structure.

**Comment by extrowerk** [comment]
Here is a complete dependency list, please fix your REQ/BUILD_REQ's:
```
  NEEDED               libQt5PrintSupport.so.5
  NEEDED               libQt5Widgets.so.5
  NEEDED               libQt5Gui.so.5
  NEEDED               libQt5Sql.so.5
  NEEDED               libQt5Core.so.5
  NEEDED               libGL.so.1
```

**Comment by mazbrili** [comment]
> Here is a complete dependency list, please fix your REQ/BUILD_REQ's:
> 
> ```
>   NEEDED               libQt5PrintSupport.so.5
>   NEEDED               libQt5Widgets.so.5
>   NEEDED               libQt5Gui.so.5
>   NEEDED               libQt5Sql.so.5
>   NEEDED               libQt5Core.so.5
>   NEEDED               libGL.so.1
> ```

ok.. update recipe

File: app-text/mtafsir/mtafsir-0.2.recipe [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Sql$secondaryArchSuffix
+	devel:libQt5PrintSupport$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:qmake >= 5
```
Comment (Begasus) [inline app-text/mtafsir/mtafsir-0.2.recipe]: Needs secondaryArchSuffix:
```diff
- 	cmd:qmake >= 5
+ 	cmd:qmake$secondaryArchSuffix >= 5
```

File: app-text/mtafsir/mtafsir-0.2.recipe [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+BUILD()
+{
+	qmake
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/Mtafsir/{bin,share}
+	mkdir -p $dataDir/ayat
```
Comment (Begasus) [inline app-text/mtafsir/mtafsir-0.2.recipe]: Combine the two:
```diff
-	mkdir -p $appsDir/Mtafsir/{bin,share}
-	mkdir -p $dataDir/ayat
+	mkdir -p $appsDir/Mtafsir/{bin,share} $dataDir/ayat
```

File: app-text/mtafsir/mtafsir-0.2.recipe [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	cp -rf ./usr/bin/mtafsir $appsDir/Mtafsir/bin/Mtafsir
+	cp -rf ./usr/share/* $appsDir/Mtafsir/share
+	rm -rf $appsDir/Mtafsir/share/mtafsir/tafasir/{tafasirInfo,baghawy.db,katheer.db,qortoby.db,sa3dy.db,tabary.db,tafheem.db}
+	cp -rf $appsDir/Mtafsir/share/mtafsir/tafasir $dataDir/ayat/
+	rm -rf $appsDir/Mtafsir/share/mtafsir/tafasir
+	ln -s  /system/data/ayat/tafasir $appsDir/Mtafsir/share/mtafsir/tafasir
```
Comment (Begasus) [inline app-text/mtafsir/mtafsir-0.2.recipe]: Use $dataDir too here:
```diff
- 	ln -s  /system/data/ayat/tafasir $appsDir/Mtafsir/share/mtafsir/tafasir
+ 	ln -s  $dataDir/ayat/tafasir $appsDir/Mtafsir/share/mtafsir/tafasir
```

File: app-text/mtafsir/mtafsir-0.2.recipe [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+INSTALL()
+{
+	mkdir -p $appsDir/Mtafsir/{bin,share}
+	mkdir -p $dataDir/ayat
+	cp -rf ./usr/bin/mtafsir $appsDir/Mtafsir/bin/Mtafsir
+	cp -rf ./usr/share/* $appsDir/Mtafsir/share
+	rm -rf $appsDir/Mtafsir/share/mtafsir/tafasir/{tafasirInfo,baghawy.db,katheer.db,qortoby.db,sa3dy.db,tabary.db,tafheem.db}
+	cp -rf $appsDir/Mtafsir/share/mtafsir/tafasir $dataDir/ayat/
+	rm -rf $appsDir/Mtafsir/share/mtafsir/tafasir
```
Comment (Begasus) [inline app-text/mtafsir/mtafsir-0.2.recipe]: Remove icons folder also (not needed):
```diff
- 	rm -rf $appsDir/Mtafsir/share/mtafsir/tafasir
+ 	rm -rf $appsDir/Mtafsir/share/{icons,mtafsir/tafasir}
```

**Pull request merged**